src/java/org/apache/cassandra/service/GCInspector.java

```diff
@@ -357,18 +405,63 @@ public void setGcLogThresholdInMs(long threshold)
         if (gcWarnThresholdInMs != 0 && threshold > gcWarnThresholdInMs)
             throw new IllegalArgumentException("Threshold must be less than gcWarnThresholdInMs which is currently "
                                                + gcWarnThresholdInMs);
-
-        DatabaseDescriptor.setGCLogThreshold((int) threshold);
+        DatabaseDescriptor.setZGCLogThreshold(threshold);
+        DatabaseDescriptor.setGCLogThreshold(threshold);
     }
 
+    @Override
     public long getGcLogThresholdInMs()
     {
-        return DatabaseDescriptor.getGCLogThreshold();
+        return hasZGC ? DatabaseDescriptor.getZGCLogThreshold() : DatabaseDescriptor.getGCLogThreshold();
     }
 
+    @Override
     public long getStatusThresholdInMs()
     {
         return getGcWarnThresholdInMs() != 0 ? getGcWarnThresholdInMs() : getGcLogThresholdInMs();
     }
 
+    @Override
+    public void setGcPauseLogThresholdInMs(long threshold)
+    {
+        if (threshold <= 0)
```

Comment:
The `DurationSpec` class verifies only bigger than 0, and this method is used only by JMX? If it is the source of truth, then I think we miss to check for this setting not being set to 0 in cassandra.yaml. The check should be in the DatabaseDescriptor, happening during startup.